context.go

```diff
@@ -289,39 +289,67 @@ func normalizeFlags(flags []Flag, set *flag.FlagSet) error {
 
 type requiredFlagsErr interface {
 	error
-	getMissingFlags() []string
+	getMissingDefaultFlags() []string
+	getMissingCustomFlags() []string
 }
 
 type errRequiredFlags struct {
-	missingFlags []string
+	missingDefaultFlags []string
+	missingCustomFlags  []string
 }
 
 func (e *errRequiredFlags) Error() string {
-	numberOfMissingFlags := len(e.missingFlags)
-	if numberOfMissingFlags == 1 {
-		return fmt.Sprintf("Required flag %q not set", e.missingFlags[0])
+	var allErrors []string
+	numberOfMissingFlags := len(e.missingDefaultFlags)
+	numberOfMissingReqErrFlags := len(e.missingCustomFlags)
+
+	if numberOfMissingFlags > 0 {
+		if numberOfMissingFlags == 1 {
+			allErrors = append(allErrors, fmt.Sprintf("Required flag %q not set", e.missingDefaultFlags[0]))
+		} else {
+			joinedMissingFlags := strings.Join(e.missingDefaultFlags, ", ")
+			allErrors = append(allErrors, fmt.Sprintf("Required flags %q not set", joinedMissingFlags))
+		}
 	}
-	joinedMissingFlags := strings.Join(e.missingFlags, ", ")
-	return fmt.Sprintf("Required flags %q not set", joinedMissingFlags)
+
+	if numberOfMissingReqErrFlags > 0 {
+		for i := range e.missingCustomFlags {
+			allErrors = append(allErrors, e.missingCustomFlags[i])
+		}
+	}
+
+	return strings.Join(allErrors, "\n")
 }
 
-func (e *errRequiredFlags) getMissingFlags() []string {
-	return e.missingFlags
+func (e *errRequiredFlags) getMissingDefaultFlags() []string {
+	return e.missingDefaultFlags
+}
+
+func (e *errRequiredFlags) getMissingCustomFlags() []string {
+	return e.missingCustomFlags
 }
 
 func checkRequiredFlags(flags []Flag, context *Context) requiredFlagsErr {
-	var missingFlags []string
+	var missingDefaultFlags []string
+	var missingCustomFlags []string
```

Comment:
❓ _(blocking question)_

Can you help me understand the motivation behind separating `missingDefaultFlags` and `missingCustomFlags`? Thinking apart this functionality conceptually, I'm not sure that separation is strictly required.

Reply:
Yeah, that was just the way I came up with to allow for both default and custom flag errors in the same app. It does seem longwinded. I'd be glad to update it if you have suggestions.

Reply:
The default and custom errors can exist within the same app if you always use the default error when the custom error is undefined.

So instead of having this

```golang
      Required: true, 
      RequiredFlagErr: cli.FlagErr{
        Custom: true, 
        Message: `There's a problem: "lang" is a required flag` ,
      },
```

You have this

```golang
      Required: true, 
      RequiredFlagErr: `There's a problem: "lang" is a required flag`,
```

Although ideally you would name the attribute based on user perception, so something like so

```golang
      Required: true, 
      RequiredErrorMessage: `There's a problem: "lang" is a required flag`,
```

So my _must change_ request here is, start with this api ^ and then work on getting that code change working with everything else.